test/SurveyDesign.jl

```diff
@@ -14,6 +14,25 @@
     # TODO: needs change; this works but isn't what the user is expecting
     srs_prob = SimpleRandomSample(apisrs; probs = fill(0.3, size(apisrs_original, 1)))
     @test srs_prob.data.probs[1] == 0.3
+    @test srs_prob.data.weights == 1 ./ srs_prob.data.probs
+
+    srs = SimpleRandomSample(apisrs, ignorefpc = true, weights = :pw )
+    @test srs.data.probs == 1 ./ srs.data.weights
+    
+    srs = SimpleRandomSample(apisrs, ignorefpc = false, weights = :pw)
+    @test srs.data.probs == 1 ./ srs.data.weights
+
+    srs_weights = SimpleRandomSample(apisrs, ignorefpc = false, weights = :fpc)
+    @test_throws srs_weights = SimpleRandomSample(apisrs, ignorefpc = false, weights = :stype)
+    srs_w_p = SimpleRandomSample(apisrs, ignorefpc = false, weights = :fpc, probs = fill(0.3, size(apisrs_original, 1)))
+
+    srs = SimpleRandomSample(apisrs, ignorefpc = true, probs = 1 ./ apisrs.pw )
+    @test srs.data.probs == 1 ./ srs.data.weights
+    srs = SimpleRandomSample(apisrs, popsize = -2.8, ignorefpc = true)# the errror is wrong
+    srs = SimpleRandomSample(apisrs, sampsize = -2.8, ignorefpc = true)# the function is working upto line 55
```

Comment:
These last two lines give errors. There are no tests attached to them. We should check that these lines do indeed throw an error with `@test_throws ERROR_NAME`. Otherwise, they will simply throw errors (uncaught) and stop the process. If the error thrown right now is wrong, maybe we should wait to do the merge after we fix the error.